addons/mail/static/src/components/activity/activity.js

```diff
@@ -175,7 +175,7 @@ class Activity extends Component {
      * @param {MouseEvent} ev
      */
     _onClickEdit(ev) {
-        this.activity.edit();
+        this.activity.edit(() => this.trigger('reload', { keepChanges: true }));
```

Comment:
alternative to passing a callback would be for edit to return a promise that is resolved on close and using await here